### PR TITLE
rustdoc-gui: allow running on Windows

### PR DESCRIPTION
**Pull request by klensy**
This adds few fixes to allow running `python x.py test rustdoc-gui` on Windows.

* path to npm required to be `npm.cmd` on Windows (otherwise don't work for me)
* properly parse node module version on Windows
* properly provide path to browser-ui-test runner (fixed in #112613)

r? @GuillaumeGomez

### CODE REVIEW

File: src/tools/rustdoc-gui-test/src/main.rs [hunk 1]
```diff
@@ -20,7 +20,7 @@ fn get_browser_ui_test_version_inner(npm: &Path, global: bool) -> Option<String>
         .unwrap_or(String::new());
     lines
         .lines()
-        .find_map(|l| l.split(':').nth(1)?.strip_prefix("browser-ui-test@"))
+        .find_map(|l| l.rsplit(':').next()?.strip_prefix("browser-ui-test@"))
```
Comment (klensy) [inline src/tools/rustdoc-gui-test/src/main.rs:23]: Consider windows path `C:\\proj\\rust\\node_modules\\browser-ui-test:browser-ui-test@0.16.6`: in that case `split` will find wrong part of string.

**Comment by klensy** [comment]
Some help required here:https://github.com/rust-lang/rust/blob/cb882fa998571e8a7ec1c06bb5d9dd9bc3423629/src/tools/rustdoc-gui/tester.js#L204

This on windows will be like `C:\proj\rust\build\x86_64-pc-windows-msvc\test\rustdoc-gui\doc` and later converted to `C:/projust%08uild%C2%86_64-pc-windows-msvcestustdoc-guidoc`, wrongly processing escapes. Can this be fixed on js side?

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
> Some help required here:
> 
> https://github.com/rust-lang/rust/blob/cb882fa998571e8a7ec1c06bb5d9dd9bc3423629/src/tools/rustdoc-gui/tester.js#L204
> 
> This on windows will be like `C:\proj\rust\build\x86_64-pc-windows-msvc\test\rustdoc-gui\doc` and later converted to `C:/projust%08uild%C2%86_64-pc-windows-msvcestustdoc-guidoc`, wrongly processing escapes. Can this be fixed on js side?

Oh interesting. I need to check if it's something done wrong in `src/tools/rustdoc-gui/tester.js` or directly into `browser-ui-test`. I'll do that in the next hours.

**Comment by klensy** [comment]
Added little more info to error description when calling npm, previously error was silently skipped.

**Pull request merged**